config/keys.go

```diff
@@ -65,5 +71,27 @@ func LoadSigningKeys() (*SigningKeys, error) {
 		}
 		return nil, err
 	}
+
+	if err := validateKeys(&config); err != nil {
+		return nil, err
+	}
+
 	return &config, nil
 }
+
+func validateKeys(config *SigningKeys) error {
+	keys := config.Keys
+	uniqMap := make(map[string]struct{}, len(keys))
+	for _, key := range keys {
+		if _, ok := uniqMap[key.Name]; ok {
+			return fmt.Errorf("malformed %s: multiple keys with name '%s' found", dir.PathSigningKeys, key.Name)
+		}
+		uniqMap[key.Name] = struct{}{}
+	}
+
+	if _, ok := uniqMap[config.Default]; !ok {
```

Comment:
IIRC, config.Default can be an empty string in the signingkeys.json file. So, missing config.Default should be fine here?

Reply:
Fixed